File: SSUET Projects/ToDoApp SqlLite/app/src/main/java/com/example/farhan/todoappSqlLite/MainActivity.java
package com.example.farhan.todoappSqlLite;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ListView;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private EditText editText;
    private ArrayList<DataSource> dataSources;
    CustomArrayAdapter customArrayAdapter;
    private List<DataSource> contactsDb;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        dataSources = new ArrayList<>();

        editText = (EditText) findViewById(R.id.etSearch);
        Button btnInsert = (Button) findViewById(R.id.btnInsert);

        ListView listView = (ListView) findViewById(R.id.mListView);

        customArrayAdapter = new CustomArrayAdapter(this, dataSources, editText);
        listView.setAdapter(customArrayAdapter);

        doAddDataInArrayList();

        btnInsert.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                String etFieldData = editText.getText().toString();

                if(!etFieldData.isEmpty()){

                    DataSource ds = new DataSource(etFieldData);
                    ds.save();

                    Toast.makeText(MainActivity.this, "Data Inserted", Toast.LENGTH_SHORT).show();

                    doAddDataInArrayList();
                    customArrayAdapter.notifyDataSetChanged();
                }else {
                    editText.setError("This Field Can't Be Empty");
                }

            }
        });

    }

    // Getting all data from Class and stores in LIst to show.
    private void doAddDataInArrayList() {
        contactsDb = DataSource.listAll(DataSource.class);
        dataSources.clear();
        dataSources.addAll(contactsDb);

    }

}

File: My Practice Projects/PracticeMapSearchByPlaceName/app/src/main/java/example/farhan/com/practicemapsearchbyplacename/MainActivity.kt
package example.farhan.com.practicemapsearchbyplacename

import android.location.Address
import android.location.Geocoder
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.widget.Toast
import java.io.IOException


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val coder = Geocoder(this)
        val address: List<Address>
        try {

            address = coder.getFromLocationName("gulshan", 5)
            if (address == null) {
                Toast.makeText(this, "No Location is found", Toast.LENGTH_LONG).show()
                Log.e("TAG", "address == null")
            }

            Log.e("TAG", " size ${address.size}")
            val location = address[0]

            val lat = location.latitude
            val lon = location.longitude

            Log.e("TAG", "\"Latitude $lat and Longitude $lon\"")
            Toast.makeText(this, "Latitude $lat and Longitude $lon", Toast.LENGTH_LONG).show()
        } catch (e: IOException) {
            e.printStackTrace()
        }

    }
}
